src/events/eventToAction.js

```diff
@@ -318,7 +318,13 @@ export default (state: PerAccountState, event: $FlowFixMe): EventAction | null =
         // TODO(server-4.0): Simplify to just use `op`.
         op: event.op ?? event.operation,
 
+        message_details: event.message_details
+          ? new Map(Object.entries(event.message_details).map(([k, v]) => [parseInt(k, 10), v]))
+          : undefined,
+
         allMessages: state.messages,
+
+        ownUserId: getOwnUserId(state),
```

Comment:
As I mentioned in another comment just above, the existing `allMessages` thing is kind of a hack.

We have a cleaner solution now, which I introduced in a2000b9c8 and we've applied in a few places since then: the sub-reducers can just take the whole state (specifically the `PerAccountState`, since introducing that distinction) as a third argument. Then they can apply a selector to get whatever particular data they need.